src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -1710,7 +1762,7 @@ private bool SendCloseResponseOnCloseIfRequired(CancellationToken token)
                 {
                     try
                     {
-                        SendCloseResponse(pendingCloseRequestContext, pendingCloseResponse, token);
+                        await SendCloseResponseAsync(pendingCloseRequestContext, pendingCloseResponse, token);
                         // this.inputSessionClosedHandle.Set();
                         cleanupCloseState = false;
```

Comment:
Is there a reason why the synchronization component of the two way close isn't being used? Without this you can have race conditions. There's a class called AsyncManualResetEvent in the codebase which might be a good fit for replacing the handle with.

Reply:
This code path is for Http (not net TCP), I am not sure two way close is being used here... if I am not wrong this was used in http path for ReliableSession class to receive.  While I have incorporated the two way close for NetTCP, I don't think we need here for http path.